packages/scheduler/lib/scheduler.ts

```diff
@@ -163,12 +163,13 @@ export class Scheduler {
     /**
      * Fail a task
      * @param taskId - Task ID
+     * @param error - Json object representing the error
      * @returns Task
      * @example
-     * const failed = await scheduler.fail({ taskId: '00000000-0000-0000-0000-000000000000' });
+     * const failed = await scheduler.fail({ taskId: '00000000-0000-0000-0000-000000000000', error: {message: 'error'});
      */
-    public async fail({ taskId }: { taskId: string }): Promise<Result<Task>> {
-        const failed = await tasks.transitionState({ taskId, newState: 'FAILED' });
+    public async fail({ taskId, error }: { taskId: string; error: JsonValue }): Promise<Result<Task>> {
```

Comment:
error format is not enforced by the scheduler. The business logic (aka server and the processor) can define their own format. It is less prescriptive but allow for more flexibility.